cpp/src/parquet/size_statistics_test.cc

```diff
@@ -0,0 +1,307 @@
+  inline static const PageSizeStatistics kEmptyPageStats{};
+};
+
+TEST_F(SizeStatisticsRoundTripTest, DisableSizeStats) {
+  auto schema = ::arrow::schema({
+      ::arrow::field("a", ::arrow::list(::arrow::list(::arrow::int32()))),
+      ::arrow::field("b", ::arrow::list(::arrow::list(::arrow::utf8()))),
+  });
+  WriteFile(SizeStatisticsLevel::None, ::arrow::TableFromJSON(schema, {R"([
+      [ [[1],[1,1],[1,1,1]], [["a"],["a","a"],["a","a","a"]] ],
+      [ [[0,1,null]],        [["foo","bar",null]]            ],
+      [ [],                  []                              ],
+      [ [[],[null],null],    [[],[null],null]                ]
+    ])"}));
+
+  ReadSizeStatistics();
+  EXPECT_THAT(row_group_stats_,
+              ::testing::ElementsAre(kEmptyRowGroupStats, kEmptyRowGroupStats,
+                                     kEmptyRowGroupStats, kEmptyRowGroupStats));
+  EXPECT_THAT(page_stats_, ::testing::ElementsAre(kEmptyPageStats, kEmptyPageStats,
+                                                  kEmptyPageStats, kEmptyPageStats));
+}
+
+TEST_F(SizeStatisticsRoundTripTest, EnableColumnChunkSizeStats) {
+  auto schema = ::arrow::schema({
+      ::arrow::field("a", ::arrow::list(::arrow::list(::arrow::int32()))),
+      ::arrow::field("b", ::arrow::list(::arrow::list(::arrow::utf8()))),
+  });
+  WriteFile(SizeStatisticsLevel::ColumnChunk, ::arrow::TableFromJSON(schema, {R"([
+      [ [[1],[1,1],[1,1,1]], [["a"],["a","a"],["a","a","a"]] ],
+      [ [[0,1,null]],        [["foo","bar",null]]            ],
+      [ [],                  []                              ],
+      [ [[],[null],null],    [[],[null],null]                ]
+    ])"}));
+
+  ReadSizeStatistics();
+  EXPECT_THAT(row_group_stats_,
+              ::testing::ElementsAre(SizeStatistics{/*ref_levels=*/{2, 2, 5},
+                                                    /*def_levels=*/{0, 0, 0, 0, 1, 8},
+                                                    /*byte_array_bytes=*/std::nullopt},
+                                     SizeStatistics{/*ref_levels=*/{2, 2, 5},
+                                                    /*def_levels=*/{0, 0, 0, 0, 1, 8},
+                                                    /*byte_array_bytes=*/12},
+                                     SizeStatistics{/*ref_levels=*/{2, 2, 0},
+                                                    /*def_levels=*/{0, 1, 1, 1, 1, 0},
+                                                    /*byte_array_bytes=*/std::nullopt},
+                                     SizeStatistics{/*ref_levels=*/{2, 2, 0},
+                                                    /*def_levels=*/{0, 1, 1, 1, 1, 0},
+                                                    /*byte_array_bytes=*/0}));
+  EXPECT_THAT(page_stats_, ::testing::ElementsAre(kEmptyPageStats, kEmptyPageStats,
+                                                  kEmptyPageStats, kEmptyPageStats));
+}
+
+TEST_F(SizeStatisticsRoundTripTest, EnablePageSizeStats) {
```

Comment:
So the three tests here use the exact same data and scaffolding with some minor variations, how about factoring things out? (perhaps even as a single test)